src/request/sign-polygon-transaction/SignPolygonTransaction.js

```diff
@@ -0,0 +1,235 @@
+/* global ethers */
+/* global Key */
+/* global KeyStore */
+/* global PolygonContractABIs */
+/* global PasswordBox */
+/* global Errors */
+/* global Utf8Tools */
+/* global TopLevelApi */
+/* global PolygonAddressInfo */
+/* global NumberFormatting */
+/* global CONFIG */
+/* global PolygonKey */
+/* global OpenGSN */
+
+/**
+ * @callback SignPolygonTransaction.resolve
+ * @param {KeyguardRequest.SignedPolygonTransaction} result
+ */
+
+class SignPolygonTransaction {
+    /**
+     * @param {Parsed<KeyguardRequest.SignPolygonTransactionRequest>} request
+     * @param {SignPolygonTransaction.resolve} resolve
+     * @param {reject} reject
+     */
+    constructor(request, resolve, reject) {
+        this._request = request;
+        /** @type {HTMLElement} */
+        this.$el = (document.getElementById(SignPolygonTransaction.Pages.CONFIRM_TRANSACTION));
```

Comment:
In all of those situations if we are sure that the element exist we can cast the returned value of `getElementById` to be `HTMLElement` instead of `HTMLElement|null` which currently conflict with the type definition above it.
```suggestion
        /** @type {HTMLElement} */
        this.$el = /**@type {HTMLElement}*/(document.getElementById(SignPolygonTransaction.Pages.CONFIRM_TRANSACTION));
```
Or if we're not sure the element exist, we can also specify it by adding a `?` before the type, which also fix the type conflict.
```suggestion
        /** @type {?HTMLElement} */
        this.$el = document.getElementById(SignPolygonTransaction.Pages.CONFIRM_TRANSACTION));
```

Reply:
AFAIK this conflict is only in our IDEs, since it uses a newer TS version than the Keyguard code.
It compiles correctly.

This will be part of the changes required when updating the Keyguard's TS version.